tree/account_tree.go

```diff
@@ -52,16 +52,19 @@ func InitAccountTree(
 	opts := ctx.Options(blockHeight)
 
 	// init account state trees
-	accountAssetTrees = make([]bsmt.SparseMerkleTree, accountNums)
+	accountAssetTrees = NewLazyTreeCache(types.AssetTreeCacheSize)
 	for index := int64(0); index < accountNums; index++ {
 		// create account assets tree
-		accountAssetTrees[index], err = bsmt.NewBASSparseMerkleTree(bsmt.NewHasher(mimc.NewMiMC()),
-			SetNamespace(ctx, accountAssetNamespace(index)), AssetTreeHeight, NilAccountAssetNodeHash,
-			opts...)
-		if err != nil {
-			logx.Errorf("unable to create new tree by assets: %s", err.Error())
-			return nil, nil, err
-		}
+		accountAssetTrees.AddToIndex(index, func() bsmt.SparseMerkleTree {
```

Comment:
After this function the size of `funcMap` will be `accountNums`, however `accountNums` can be huge, please come up with a solution to limit the size of `funcMap` as well.